### PR TITLE
dont send createonly to the API

### PR DESCRIPTION
**Pull request by philbrookes**
the createOnly flag had been added to the API object, which was breaking the provision of createonly realms.

To verify this:
```
oc new-project sso-test
oc create -f deploy/rbac.yaml 
oc create -f deploy/operator.yaml 
oc create -f deploy/examples/keycloak_min.json 
oc create -f deploy/examples/keycloakRealm.json 
```
- See the realm is created correctly with users.
- Set a firstname for the users in the CR and save it
- See it is not reconciled into the user
- Delete the createOnly from the CR
- See the first name is reconciled.

### CODE REVIEW

**Comment by maleck13** [comment]
I should probably change something about the realm and see it doesn't get set back too

**Comment by philbrookes** [comment]
I've been using the users firstname, but try anything you like, just the roles in users are malfucntioning, so dont use them

File: deploy/operator.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ spec:
     spec:
       containers:
         - name: keycloak-operator
-          image: quay.io/integreatly/keycloak-operator:master
+          image: quay.io/integreatly/keycloak-operator:v1.2.4
```
Comment (philbrookes) [inline deploy/operator.yaml]: this needs to be 1.3.1

File: pkg/keycloak/realm/phaseHandler.go [hunk 2]
```diff
@@ -237,23 +228,28 @@ func (ph *phaseHandler) reconcileUser(kcUser, specUser *v1alpha1.KeycloakUser, r
...
+			if !resourcesEqual(kcUser.KeycloakApiUser, specUser.KeycloakApiUser) {
+				specUser.ID = kcUser.ID
+				err := authenticatedClient.UpdateUser(specUser, realmName)
+				if err != nil {
+					return err
+				}
 			}
+		} else {
+			// Need to set the ID because specUser is used for queries
+			specUser.ID = kcUser.ID
```
Comment (philbrookes) [inline pkg/keycloak/realm/phaseHandler.go]: is this an else to to the `if !createOnly`? if so, let's just put this above it? as it applies in both cases, right?

Comment (pb82) [reply pkg/keycloak/realm/phaseHandler.go]: Right, and i think this even applies regardless of the value of `createOnly` as the ID is subsequently required for listing users. Will put it into above the whole if and retest.

**Comment by philbrookes** [comment]
I can't approve this as github thinks I opened it, I have verified it on my cluster though.

**Pull request merged**